tests/conftest.py

```diff
@@ -166,75 +200,143 @@ def fixture_t4c_exporter_env(
 
 
 @pytest.fixture(name="init_t4c_server_repo")
-def fixture_init_t4c_server_repo(t4c_http_port: str):
-    create_t4c_repository(t4c_http_port)
+def fixture_init_t4c_server_repo(t4c_ip_addr: str, t4c_http_port: str):
+    create_t4c_repository(t4c_ip_addr, t4c_http_port)
     yield
 
 
 @contextlib.contextmanager
 def get_container(
     image: str,
+    image_tag_env: str | None = None,
     ports: dict[str, int | None] | None = None,
     environment: dict[str, str] | None = None,
     volumes: dict[str, dict[str, str]] | None = None,
     entrypoint: list[str] | None = None,
 ) -> cabc.Iterator[containers.Container]:
     docker_prefix = os.getenv("DOCKER_PREFIX", "")
-    docker_tag = os.getenv("DOCKER_TAG", "latest")
+    if image_tag_env and (
+        image_tag_env_value := os.getenv(image_tag_env, None)
+    ):
+        docker_tag = image_tag_env_value
+    else:
+        docker_tag = os.getenv("DOCKER_TAG", "latest")
+
     image = f"{docker_prefix}{image}:{docker_tag}"
 
     container = client.containers.run(
         image=image,
-        detach=True,
         ports=ports,
         environment=environment,
         volumes=volumes,
         entrypoint=entrypoint,
+        network=DOCKER_NETWORK if DOCKER_NETWORK != "host" else None,
+        labels={"capella-dockerimages-pytest-container": "true"},
+        detach=True,
+        tty=True,
     )
 
     try:
         yield container
     finally:
-        if container:
-            container.stop()
-            container.remove()
+        container.stop()
+        container.remove()
+
 
+def wait_for_container(
+    container: containers.Container,
+    wait_for_message: str,
+    cmd: str | None = None,
+):
+    start_time = time.time()
 
-def wait_for_container(container: containers.Container, wait_for_message: str):
-    log_line = 0
-    for _ in range(int(timeout / 2)):
-        log.info("Wait until %s", wait_for_message)
+    log_queue: multiprocessing.Queue = multiprocessing.Queue()
+
+    create_log_process = multiprocessing.Process(
+        target=create_log, args=(container.id, log_queue, cmd)
+    )
 
-        splitted_logs = container.logs().decode().splitlines()
-        log.debug("Current log: %s", "\n".join(splitted_logs[log_line:]))
-        log_line = len(splitted_logs)
+    container.reload()
+    if container.status == "exited":
+        log.error("Log from container:\n%s", container.logs().decode())
+        raise RuntimeError("Container exited unexpectedly")
```

Comment:
Unfortunately, that's not possible. This is primarily due to the limitation that we cannot directly provide the container to the log capturing process. Instead, we have to provide the container ID and then retrieve the corresponding container. As a result, there is a potential issue where the container might have already failed, leading to an error in the `create_log` function because it cannot find the container with the provided ID.

Reply:
Makes sense. Let's add a short comment why it is needed then.